### ISSUE TITLE
Add provider_id to calendar model

### ISSUE THREAD

**Issue by MelissaAutumn** [thunderbird/appointment#394]
For an easy way to grab all calendars by provider x we might want to tag the calendars with the provider's id. 

This comes up with disconnecting Google Accounts but we could similarly use it when we support other calendar providers, or even caldav calendars.

**Comment by davinotdavid** [comment]
@MelissaAutumn given that the `calendar.provider` is an `enum`, would the connection from `calendar` to `external_connection` solve it?

https://github.com/thunderbird/appointment/pull/1078/files#diff-74a1ee7069829ca2c7a235dfe2e3f383a9119bf6cf89bcdd2a784daa95f89d76R235-R465

**Comment by MelissaAutumn** [comment]
Yep. I meant to add that to my pr comments. Thanks for catching this ticket!

**Comment by MelissaAutumn** [comment]
This can be closed now.
